### PR TITLE
Much faster build, and simplified CmakeFiles.txt for tests

### PR DESCRIPTION
**Pull request by patriotyk**
Optimized build process because it is to boring, also simplified CmakeLists.txt

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -79,161 +70,87 @@ find_package(Boost COMPONENTS system thread program_options log log_setup regex
 #enable test features of cmake
 enable_testing()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -std=c++0x")
```
Comment (cajun-rat) [inline tests/CMakeLists.txt]: Is this needed?

Comment (patriotyk) [inline tests/CMakeLists.txt]: Yes, You are right, now we don't need it

**Comment by patriotyk** [comment]
There was a c++11 specific code, I have fixed it

**Comment by cajun-rat** [comment]
Nice job!  This doubles the speed of the build on my machine!

**Pull request merged**